src/measurementaudiothread.cpp

```diff
@@ -49,8 +49,14 @@ void MeasurementAudioThread::selectDevice(const QAudioDeviceInfo &deviceInfo, bo
             m_chanelCount = formatChanels;
         m_maxChanelCount = std::max(formatChanels, m_maxChanelCount);
     }
-
-    m_format.setSampleRate(48000);
+    //get current device and check avalible sample rate
+    if(deviceInfo.supportedSampleRates().last() < 48000)
```

Comment:
There is not any guarantee that _QList_ will be ordered, also audio card could support 88.2KHz and does not support 48KHz.
Check with _QList::indexOf_ will work better.
Also, you could use _QAudioDeviceInfo::isFormatSupported_ to check if 48KHz is not supported by an audio card.